js/src/elements/form-behavior.ts

```diff
@@ -396,14 +366,7 @@ class FormBehaviorElement extends HTMLElement {
 
 			const isCustomElement = !!customElements.find(el => el === input);
 
-			this.runBasicValidation(
-				input,
-				displayValidity,
-				errors,
-				errorList,
-				isTagSelector,
-				isCustomElement
-			);
+			this.runBasicValidation(input, displayValidity, errors, errorList, isCustomElement);
 		}
 
```

Comment:
I think you should fire the form-validating event here, this way the external code has an opportunity to insert validation errors into the errors array prior to us checking `if (errors.length === 0)` on the next line. This should also remove the need to dispatch a clear-validation-errors event.

Reply:
Moved the event firing. I think we still need a way to signal when to clear the custom errors though. For normal inputs, we're clearing the errors on input and change events, so it's before resubmission of the form, hence before the `form-validating` event fires again. I think we should keep that the same even for elements with custom validation. I think it would become confusing to show a stale error message if we wait to clear it until resubmission and revalidation for these custom validation inputs, but have a different behavior for normal inputs.

Tags involve hidden inputs and sibling elements, so in the external code, I'll need to re-implement the tag specific code that I'm removing in this PR.

Reply:
totally makes sense- would it be resolved if the tag control fired change events when it's value changed?

Reply:
The existing tag control in Docs does fire change events on value change so it will trigger the clearing function. The clearing function no longer handles tag, but with the custom event, I can listen for it and clear the error for tags or any other custom input. It didn't make sense to keep the tag specific error handling around when we're removing the tag validation altogether.